### ISSUE TITLE
change QC for measurements to deal with ><-

### ISSUE THREAD

**Issue by kmexter** [emo-bon/data-quality-control-action#3]
There are values in the mesurements tab that are not floats, but include a < value (a maximum, because it means that this is the lowest value that the instrument could record) 
And in principle some may also have a > (a minumum) and - (a range), tho I do not expect these

These have to be deal with in the following way
- for the QC: still report a problem
- for the transformed logsheets: copy over as-is - this means that ALL measurements have to be recorded as strings in logsheets_schema_extended 
- for the ttl - @laurianvm or @bulricht to change the procedure so that: If there is a "< float" in the cell, put the float as the value in the ttl for that parameter, and indicate that it is a maximum value; if there is a "> float" then put the float as the value and indicate that it is a minumum value; if there is a "float - float" then record it as a range min to max. 
- We will need to document that this is done because it will mean that not all values for measurements are described in the same way and the VRE code that may want to do maths on these data will run into problems. To consider: perhaps make ALL values at least a range, from min to max?  
- for the ENA workflow, as it will not accept strings, we will have to NOT include that value in the xml file BUT we will have to instead create a new entry called "[measurement name] additional info" and copy the string into there. See https://github.com/emo-bon/ena-sample-registration-action/issues/3

**Comment by kmexter** [comment]
For point 2 - with Laurian
For point 3 - point that out in the advice I will give to Bram on the ENA workflow

**Comment by laurianvm** [comment]
@kmexter and @marc-portier, checking in - is this GH issue still relevant? 
I’d like to determine if this needs to be addressed and, if so, clarify next steps and responsibilities

**Comment by kmexter** [comment]
I suspect it is still relevant - as in these need to be dealt with. But TDB whether that is done in the transformation or the ttl part (I suggested above the ttl but whichever works better is fine). @marc-portier can come to us when he gets to editing this part of the QC code

**Comment by kmexter** [comment]
@isanti: can you check to see how many of the measurements use < - or > rather than just giving a value? And whether there are any depths in the sampling tab that do that? In this way we know if this is a small or a tiny problem. I think it is a very small problem but not a zero problem - a random search of 3 of 4 logsheets and I found only limits in the nitrate column of https://docs.google.com/spreadsheets/d/1F4AWv_seI-DQJ_Gp_N2ziwvGyjnc4KC92GGNXrJRek4/edit?gid=996599872#gid=996599872, and in https://docs.google.com/spreadsheets/d/1kSlTcNfXaCQv7fIKVbBnt5PmRW3_H7Trryh4FJHyLkE/edit?gid=753216632#gid=753216632 depth is given as a range. 

The decision was made that where this occurred we would keep the information and deal with it by recording them as max and min values in the triples. (only for the two size_frac columns would it be flagged as an error and need to be fixed). However, this was made while you were away, and I'd like to know, @isanti, if you agree with this. If so, @marc-portier should include it in the QC scripts and @laurianvm in the ttl files.

**Comment by isanti** [comment]
I don't know how I missed this issue. apologies! Pls see below the list and comments.

----tab measured 

- water logsheets

https://docs.google.com/spreadsheets/d/1F4AWv_seI-DQJ_Gp_N2ziwvGyjnc4KC92GGNXrJRek4/edit?gid=996599872#gid=996599872
SZN column nitrate ; can be reported as max value

https://docs.google.com/spreadsheets/d/1jymTM_4L1T4hwRvSSbZQ7NoT6YfVZz0ClXHtSIel9w0/edit?gid=15718907#gid=15718907
TZS columns nitrate and nitrite ; they added "<LOD" which means below detection limit. Not sure how to transform this. Each instrument has a certain detection limit. I added a comment for TZS to see if they have more information.

- sediment logsheets
nothing

I would say a small issue then.

----tab sampling 
the size_frac is supposed to be a range
if size_frac_low or size_frac_up includes > < or - I would say it needs to be flagged as an error
if depth includes > < or - I would say it needs to be flagged as an error


----tab observatory
if tot_depth_water_col includes > < or - I would say it needs to be flagged as an error

**Comment by kmexter** [comment]
Agree, in tab measured it is a small issue, Indeed, I forgot that "below detection limit" also needs to know what that "detection limit" is, otherwise this is ... probably a "not measured" value (since we cannot say it is a zero value). What do you think, @isanti? 
For the SZN being a max value - @laurianvm can you do that, if then onl SZN for one measurement has this, and all the others are an exact value? 

For tab sampling, size_frac I think we are dealing with that as a string (since the limits are anyway in _up and _low), so @laurianvm confirm and if so, this is not an issue. 
Noted about the depth - @bulricht is it flagged as an error if it is reported as a range? if so, nothing to do here - it is up to @isanti to then ask the station to fix that. 

For tab observatory, I dont think anyone has reported that as a range, but again @bulricht make sure this is flagged as a QC error if it is.

**Comment by laurianvm** [comment]
> For the SZN being a max value - [@laurianvm](https://github.com/laurianvm) can you do that, if then onl SZN for one measurement has this, and all the others are an exact value?

The way the template is currently structured makes it quite challenging to implement this change, and I’m not sure the benefits would outweigh the necessary modifications. You can review the template here for more context: 
https://github.com/emo-bon/observatory-profile/blob/3bb1f4cb5a897e5c0a3b7d8105a2ccdc0ff27c69/templates/sediment_measured.ldt.ttl#L27-L60 

However, if this change is important, an alternative could be that you look for a definition of nitrate that specifies it as a max nitrate value, and then update the observable_property_url for nitrate in [logsheet_schema_extended](https://github.com/emo-bon/observatory-profile/blob/main/logsheet_schema_extended.csv)

**Comment by laurianvm** [comment]
> For tab sampling, size_frac I think we are dealing with that as a string (since the limits are anyway in _up and _low), so [@laurianvm](https://github.com/laurianvm) confirm and if so, this is not an issue.

both the range and the up/low values are included. You can find more details in the template here: 
https://github.com/emo-bon/observatory-profile/blob/3bb1f4cb5a897e5c0a3b7d8105a2ccdc0ff27c69/templates/sediment_sample.ldt.ttl#L129-L161

The template does treat the range as a string, but it expects the up/low values to be numbers (specifically of type double). This is something that should be handled by the QC/transformation rules, if it’s not already in place.

**Comment by kmexter** [comment]
> > For the SZN being a max value - [@laurianvm](https://github.com/laurianvm) can you do that, if then onl SZN for one measurement has this, and all the others are an exact value?
> 
> The way the template is currently structured makes it quite challenging to implement this change, and I’m not sure the benefits would outweigh the necessary modifications. You can review the template here for more context: https://github.com/emo-bon/observatory-profile/blob/3bb1f4cb5a897e5c0a3b7d8105a2ccdc0ff27c69/templates/sediment_measured.ldt.ttl#L27-L60
> 
> However, if this change is important, an alternative could be that you look for a definition of nitrate that specifies it as a max nitrate value, and then update the observable_property_url for nitrate in [logsheet_schema_extended](https://github.com/emo-bon/observatory-profile/blob/main/logsheet_schema_extended.csv)

@isanti - how important is this? if it is important that we record the nitrate as a max value, then we will do so but then I would prefer all values are recorded as max and min values (with 95% having max=min) because this may happen again in the future. Or am I wrong?

**Comment by kmexter** [comment]
@kmexter to summarise what is done already in the ttl templates and what still need to be done in the qc workflow.
